packages/file_selector/file_selector_android/android/src/main/java/dev/flutter/packages/file_selector_android/FileUtils.java

```
@@ -0,0 +1,198 @@
+      targetDirectory.mkdir();
+      targetDirectory.deleteOnExit();
+      String fileName = getFileName(context, uri);
+      String extension = getFileExtension(context, uri);
+
+      if (fileName == null) {
+        if (extension == null) {
+          throw new IllegalArgumentException("No name nor extension found for file.");
+        } else {
+          fileName = "file_selector" + extension;
+        }
+      } else if (extension != null) {
+        fileName = getBaseName(fileName) + extension;
+      }
+
+      File file = new File(targetDirectory, fileName);
+
+      try (OutputStream outputStream = new FileOutputStream(file)) {
+        copy(inputStream, outputStream);
+        return file.getPath();
+      }
+    } catch (IOException e) {
+      // If closing the output stream fails, we cannot be sure that the
+      // target file was written in full. Flushing the stream merely moves
+      // the bytes into the OS, not necessarily to the file.
+      return null;
+    } catch (SecurityException e) {
+      // Calling `ContentResolver#openInputStream()` has been reported to throw a
+      // `SecurityException` on some devices in certain circumstances. Instead of crashing, we
+      // return `null`.
+      //
+      // See https://github.com/flutter/flutter/issues/100025 for more details.
+      return null;
+    }
+  }
+
+  /** Returns the extension of file with dot, or null if it's empty. */
+  private static String getFileExtension(Context context, Uri uriFile) {
+    String extension;
+
+    try {
+      if (uriFile.getScheme().equals(ContentResolver.SCHEME_CONTENT)) {
+        final MimeTypeMap mime = MimeTypeMap.getSingleton();
+        extension = mime.getExtensionFromMimeType(context.getContentResolver().getType(uriFile));
+      } else {
+        extension =
+            MimeTypeMap.getFileExtensionFromUrl(
+                Uri.fromFile(new File(uriFile.getPath())).toString());
+      }
+    } catch (Exception e) {
```

Is there a more scoped exception you can catch here? If not eating an exception then not logging can make for some difficult to debug code.